pages/content/posts/securecontext-webidl/securecontext-webidl.md

```diff
@@ -0,0 +1,134 @@
+
+- The [`getCurrentPosition()`](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation/getCurrentPosition) method of [`Geolocation`](https://developer.mozilla.org/en-US/docs/Web/API/Geolocation) is not marked as requiring a secure context, but the implementation of the method requires constructing a [`GeolocationPosition`](https://developer.mozilla.org/en-US/docs/Web/API/GeolocationPosition) object, and that _does_ require a secure context. So MDN marks `getCurrentPosition()` as requiring secure context.
+
+- The [`Notification`](https://developer.mozilla.org/en-US/docs/Web/API/Notification) interface is not marked secure context, but it is defined as a [powerful feature](https://notifications.spec.whatwg.org/#permissions-integration) requiring permission, and [getting permission itself requires a secure context](https://w3c.github.io/permissions/#dfn-getting-the-current-permission-state). So, again, MDN marks `Notification` as requiring secure context.
+
+In cases like this, it seems like the docs would be worse if we used Web IDL as the source of truth.
+
+### Secure context in Web IDL, not secure context in MDN
+
+This lists the other direction: features that exist in both MDN and Web IDL, that are listed as secure context in Web IDL but not in MDN. If we just use Web IDL as the source of truth, then these MDN pages would start to mark the features as secure context.
+
+We found 668 items in this category! We haven't looked at all of them, but in all the cases we have looked at, this is an error in MDN. A common source of this issue is when the interface is marked as requiring a secure context, which means all its members require secure context. Often in cases like this MDN will only mark the interface page as requiring secure context, and not the method or property pages.
+
+So for this direction, using Web IDL as the source of truth would fix hundreds of MDN pages.
+
+### Secure context in MDN, nonexistent in Web IDL
+
+In this category we have features that are documented on MDN, but that don't have corresponding items in Web IDL at all. If we use Web IDL as the source of truth, what can we say about features that don't exist in Web IDL?
+
+There are 38 pages in this category. They fall into two types: nonstandard features, and places where MDN does not directly represent IDL structures.
+
+### Nonstandard features
+
+In our study, almost all of these are features that used to be standard but were then deprecated and removed from the specification: for example, [`PaymentAddress`](https://developer.mozilla.org/en-US/docs/Web/API/PaymentAddress). Perhaps it's not so important to display secure context information for these.
+
+However, nonstandard features are generally a problem for any broader attempt to use Web IDL as a data source. It's been suggested that we could extract the browser-specific IDL and make a nonstandard companion to the webref/idl package.
```

Comment:
Great post! Note that the discussion on SecureContext in IDL is leading to [some further interesting findings in webref](https://github.com/w3c/webref/issues/1142#issuecomment-1930188979)

Just a note on this particular proposal that if OWD/MDN are willing to maintain the data, I think we would be happy to have this integrated in some way in webref (since that's likely something other webref consumers would find useful)